animations/hero-animations/index.md

```diff
@@ -0,0 +1,666 @@
+   and the same tag as the source hero.
+   It's <strong>essential that both hero objects are created with
+   the same tag</strong>, typically an object that represents the
+   underlying data. For best results, the heroes should have
+   virtually identical widget trees.
+1. Create a route that contains the destination hero.
+   The destination route defines the widget tree that exists
+   at the end of the animation.
+1. Trigger the animation by pushing the destination route on the
+   Navigator's stack. The Navigator push and pop operations trigger
+   a hero animation for each pair of heroes with matching tags in
+   the source and destination routes.
+
+Flutter calculates the tween that animates the Hero's bounds from
+the starting point to the endpoint (interpolating size and position),
+and performs the animation in an overlay.
+
+The next section describes Flutter's process in greater detail.
+
+## Behind the scenes
+
+The following describes how Flutter performs the transition from
+one route to another.
+
+<img src="images/hero-transition-0.png" alt="Before the transition the source hero appears in the source route">
+
+Before transition, the source hero waits in the source route's widget
+tree. The destination route does not yet exist, and the overlay
+is empty.
+
+---
+
+<img src="images/hero-transition-1.png" alt="The transition begins">
+
+Pushing a route to the Navigator triggers the animation.
+At t=0.0, Flutter does the following:
+
+* Lays out the destination route&mdash;off screen&mdash;using the
+  curved path as described in the Material motion spec.
+  Flutter now knows where the hero ends up.
+
+* Places the destination hero in the overlay, at the
+  same location and size as the _source_ hero.
+  Adding a hero to the overlay changes its Z-order so that
+  it appears on top of all routes.
+
+* Moves the source hero offscreen.
+
+---
+
+<img src="images/hero-transition-2.png" alt="The hero flies in the overlay to its final position and size">
+
+As the hero flies, its rectangular bounds are animated using
+[Tween&lt;Rect&gt;,](https://docs.flutter.io/flutter/animation/Tween-class.html)
+specified in Hero's
+[`createRectTween`](https://docs.flutter.io/flutter/widgets/CreateRectTween.html) property.
+By default, Flutter uses an instance of
+[MaterialRectArcTween,](https://docs.flutter.io/flutter/material/MaterialRectArcTween-class.html)
+which animates the rectangle's opposing corners along a curved path.
+(See [Radial hero animations](#radial-hero-animations)
+for an example that uses a different Tween animation.)
+
+---
+
+<img src="images/hero-transition-3.png" alt="When the transition is complete, the hero is moved from the overlay to the destination route">
+
+When the flight completes:
+
+* Flutter moves the hero widget from the overlay to the
+  destination route. The overlay is now empty.
+
+* The destination hero appears in its final position in
+  the destination route.
+
```

Comment:
I think you have to admit that the source hero is restored to its route, since you explained that it's moved to the overlay at the flight's outset.